### PR TITLE
Reboot of IsIn PR

### LINKED ISSUE #637: A Blaze equivalent for: SELECT * WHERE t.column IN list_values
I'd like to have a way to select rows from values in a given list without a Join hack. ;)

An equivalent to SQL:
SELECT * WHERE t.column IN list_values

### PR DESCRIPTION
**Pull request by cpcloud**
closes #637

### CODE REVIEW

**Comment by cpcloud** [comment]
@mrocklin @cowlicks thoughts? i'd like to merge this fairly soon

**Comment by cpcloud** [comment]
i suppose a rename of `isin` to `is_in` might be nice.

**Comment by mrocklin** [comment]
> i suppose a rename of isin to is_in might be nice.

I'm fine with isin.  Isn't this the pandas interface?

**Comment by cpcloud** [comment]
Yes, `isin` is the pandas API

File: blaze/compute/python.py [hunk 1]
```diff
@@ -490,6 +495,11 @@ def combiner(a, b):
         return grouper, binop2, combiner, tuple(inits)
 
 
+@dispatch(IsIn, Sequence)
+def compute_up(expr, data, **kwargs):
+    return map(expr._keys.__contains__, data)
```
Comment (mrocklin) [inline blaze/compute/python.py]: Doesn't rowfunc give this to us?

File: blaze/expr/collections.py [hunk 2]
```diff
@@ -425,9 +432,26 @@ def join(lhs, rhs, on_left=None, on_right=None, how='inner'):
...
+    """
+    __slots__ = '_hash', '_child', '_keys'
+
+    @property
+    def schema(self):
+        return datashape.bool_
+
+
+def isin(child, keys):
+    return IsIn(child, frozenset(keys))
```
Comment (mrocklin) [inline blaze/expr/collections.py]: I'm slightly concerned about people using an expression for `keys`.  Request informative error message.

**Comment by mrocklin** [comment]
Looks good

**Comment by cpcloud** [comment]
merging on pass

File: blaze/expr/collections.py [hunk 3]
```diff
@@ -73,6 +77,7 @@ def sort(child, key=None, ascending=True):
 
 
 class Distinct(Expr):
+
```
Comment (cowlicks) [inline blaze/expr/collections.py]: What is up with these blank lines after `class` statements?

Comment (cpcloud) [inline blaze/expr/collections.py]: I ran `autopep8` on the file

**Comment by FRidh** [comment]
Are there any plans to allow Blaze expressions as keys?
For example, in Pandas you can use a Series as input.

**Pull request merged**